fuel-block-producer/src/adapters/transaction_selector.rs

```diff
@@ -99,25 +100,23 @@ mod tests {
                 })
                 // The block producer assumes transactions are already checked
                 // so it doesn't need to compute valid sigs for tests
-                .finalize_checked_without_signature(
+                .finalize_checked_basic(
                     0,
                     &ConsensusParameters {
                         gas_price_factor: 1,
                         ..ConsensusParameters::default()
                     },
                 )
             })
+            .map(Into::into)
```

Comment:
nit: these are 3 `map()`s in a row, why not `Arc::new(checked_tx.into())` in the first map instead